src/App.js

```diff
@@ -22,25 +22,35 @@ class App extends React.Component {
       description: this.state.newTodoDescription,
       completed: false
     }
-    console.log("createTodo", newTodo)
 
-    // TODO: add the new Todo to the list in state
-    // make sure not to mutate the existing array!
+    // Add the new Todo to the list in state without mutating the original array
+    const newTodos = [...this.state.todos, newTodo]
+
+    this.setState({
+      todos: newTodos,
+      newTodoDescription: ""
+    })
   }
 
   deleteTodo = id => {
-    console.log("deleteTodo", id)
+    // Remove ONE todo from state using the id without mutating the existing array
+    const copyTodos = [...this.state.todos];
+
+    const copyTodosWithoutItem = copyTodos.filter( todo => todo.id !== id ) // only keep the todo items that do not have the id passed in
```

Comment:
Since `.filter` returns a new array, it's not necessary to make a copy with spread first - you can safely do this without mutating data:

```js
const copyTodosWithoutItem = this.state.todos.filter( todo => todo.id !== id )
```